wave_lang/kernel/wave/memory_analysis/minimize_shared_allocs.py

```diff
@@ -46,9 +46,9 @@ def compute_live_intervals(allocs: list[fx.Node]):
 
 def get_shared_memory_allocation_size(alloc: fx.Node, dtype: DataType) -> int:
     custom = get_custom(alloc)
-    return math.prod([subs_idxc(x) for x in custom.distributed_shape]) * (
-        dtype.bitwidth() // 8
-    )
+    return (
+        math.prod(subs_idxc(x) for x in custom.distributed_shape) + custom.tail_padding
```

Comment:
Is there a more ergonomic/obvious place we can insert this change into? It is not too obvious to me that it should live here

Reply:
yeah, made it an `Allocate` op method